spec/formalism/form/coercion_spec.rb

```diff
@@ -0,0 +1,189 @@
+# frozen_string_literal: true
+
+## This is unit test, created to simplify testing of different coercions
+## You can find integration test in spec/formalism/form/fields_spec.rb
+## (Formalism::Form -> .field -> coersion)
+##
+## In tests `eql` used instead of `eq` to determine result type
+describe Formalism.const_get(:Coercion) do
```

Comment:
Why not just `Formalism::Coersion`?

Reply:
It's private

Reply:
> It's private

Then String, please.

Reply:
I think we should not test private classes. So, it should be public or not tested directly. Or in another way, like through `Form`.

Reply:
> I think we should not test private classes

Why? This is exactly what unit tests for

Reply:
> Then String, please.

It will not set `described_class`. I can rewrite specs to use Form instead, but it will only make it more complicated i think. We have bunch of integration specs to ensure integration is tested

Reply:
> Why? This is exactly what unit tests for

Unit tests for unit testing. Coercion  is unit, but it's private unit. And any tests should be written through public interface in my opinion. So, for public unit directly, or for nearest public interface (parent, container, composition).

Reply:
Private is mean "you should not rely on me, I can be rewritten".

Reply:
>  And any tests should be written through public interface in my opinion. So, for public unit directly, or for nearest public interface (parent, container, composition).

In unit tests methods sometimes are overrided by stubs and private methods are called. OAs 

> Private is mean "you should not rely on me, I can be rewritten".

Unit tests are like "if you rewrite something we will fail or even not fail which is worse" :(

Ok, i understood your position, will rewrite it via form

Reply:
Removed Formialism::Coercion references from coercion_spec

Reply:
> In unit tests methods sometimes are overrided by stubs and private methods are called.

Stubbing is a different thing — "stub things (public interfaces) out of tested unit". It's not about public/private interface of tested unit.

> Unit tests are like "if you rewrite something we will fail or even not fail which is worse" :(

OK, but them should be written for public interfaces, not private.